tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsCheckpointer.cs

```diff
@@ -0,0 +1,102 @@
+// <copyright file="DataStreamsCheckpointer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Headers;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.DataStreamsMonitoring;
+
+/// <summary>
+/// Use this class to get an instance of an IDataStreamsCheckpointer.
+/// </summary>
+public class DataStreamsCheckpointer : IDataStreamsCheckpointer
+{
+    internal static readonly IDataStreamsCheckpointer Instance = new DataStreamsCheckpointer();
+
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(DataStreamsCheckpointer));
+
+    private DataStreamsCheckpointer()
+    {
+    }
+
+    /// <summary>
+    /// Get an instance of a checkpointer
+    /// </summary>
+    /// <returns>A checkpointer instance</returns>
+    public static IDataStreamsCheckpointer Get()
+    {
+        var dsm = Tracer.Instance.TracerManager.DataStreamsManager;
+        if (dsm != null && dsm.IsEnabled)
+        {
+            return Instance;
+        }
+
+        return NoopCheckpointer.Instance;
+    }
```

Comment:
This is only ever going to be called manually right? I think I'd rather have some kind of notification that DSM is disabled instead of silently noop-ing, maybe an argument exception? Or at least a log as you did with the checkpoints

Reply:
I think it's ok to call this with DSM disabled, like, if you deploy your code to N machines, but only 50% have DSM enabled, which is a pretty frequent setup I think. I can log it as info though.